### PR TITLE
fix: Use lowercase 'here' key in PJS getDryRunXcm

### LINKED ISSUE #2051: [Bug bounty] PJS SDK: relay-chain destination (Here) never detected in dry-run due to wrong key casing
## 🔍 Describe the Bug

**File:** `packages/sdk-pjs/src/PolkadotJsApi.ts`

The PJS SDK's dry-run XCM forwarding logic uses **wrong key casing** when checking for `Here` junctions (relay-chain destinations), causing it to never detect relay-chain destinations.

### Two inconsistent and both-buggy implementations of the same logic:

**Line 444** (`getDryRunCall`):
```ts
i.here === null ? 0 : ...    // WRONG: property is "Here", not "here"
```

**Line 700** (`getDryRunXcm`):
```ts
i.Here ? 0 : ...             // WRONG: i.Here is null (falsy), so truthiness check fails
```

The Substrate XCM spec encodes `Here` as `{ Here: null }`. Both implementations fail:
- Line 444 uses lowercase `here` — the property doesn't exist, so `undefined === null` is `false`
- Line 700 checks `i.Here ?` — but `i.Here` is `null` (falsy), so the ternary falls through

**Both should be:** `i.Here === null ? 0 : ...`

## 🧪 To Reproduce

The bug is encoded in the test suite — `PolkadotJsApi.test.ts:1395` asserts `destParaId: undefined` for a `{ Here: null }` relay destination, when the correct value should be `0`:

```ts
// Test fixture (line 1247): interior: { Here: null }
// Expected (encodes bug): destParaId: undefined
// Correct:               destParaId: 0
```

Compare with the **Dedot backend** (`DedotApi.ts`) which handles this correctly.

## 🖥️ Environment

- XCM Tool: XCM SDK — PJS (PolkadotJS) backend
- Component: `PolkadotJsApi` dry-run XCM forwarding
- Version: latest `develop` (commit `3c88db1`)

## 💡 Expected Behavior

Both methods should detect `{ Here: null }` junctions and return `destParaId: 0` for relay-chain destinations, consistent with the Dedot backend.

**Fix:**
- Line 444: Change `i.here` → `i.Here`
- Line 700: Change `i.Here ?` → `i.Here === null ?`

## 📍 Additional Context

The PAPI and Dedot backends both had similar issues fixed recently (#2009, #2011). The PJS backend is the last one with this class of bug. This affects dry-run accuracy for any XCM transfer with a relay-chain destination via the PJS backend.

---

## ✅ Contributor Eligibility

- [x] My GitHub account's commit history is at least 6 months old.
- [x] I have set an on-chain identity on the Polkadot People Chain for my AssetHub address (below) and requested a registrar judgement of at least `Reasonable` (verifiable via [Subscan](https://people-polkadot.subscan.io/account/16epYi2zGsDq6fsrJeJsUiFGtYfFxdHre5yPpHXdVcu84AbP)).
- [x] I disclose any AI tools used to help write this report (or confirm none were used), and I understand and can explain its contents myself.
- [x] This report was authored and submitted by a human — not opened autonomously by an AI agent.
- [x] I will respond to maintainer questions with specific, on-topic answers within 5 days.
- [x] If I go on to fix this issue myself, I understand every commit must be signed (`-S`) and include `Signed-off-by` and `AI-Assisted-By` trailers.
- [x] This is the only bug bounty issue I currently have reserved.

AI tools used: Hermes Agent (for code analysis and drafting assistance).

---

## 💸 Polkadot Asset Hub Address (Optional – for reward eligibility)

Polkadot Asset Hub Address: `16epYi2zGsDq6fsrJeJsUiFGtYfFxdHre5yPpHXdVcu84AbP`

### PR DESCRIPTION
**Pull request by yaronski**
# 🐞 Bug Fix Pull Request

## 📌 Related Issue

Closes #2051

---

## 🛠️ Description of the Fix

- **Bug description:** The PJS (PolkadotJS) API backend in `PolkadotJsApi.ts` serializes XCM junctions to lowercase keys in its `toJSON()` output (e.g., `{ interior: { here: null } }`). However, the `getDryRunXcm` method on line 700 checked `i.Here` (uppercase `H`), which **never matches** the actual serialized output. This meant relay-chain destinations (the `here` junction) were silently skipped during dry-run fee estimation, causing `destParaId` to fall through and return `undefined` instead of `0`.

  Note: As the maintainer pointed out in the issue, the correct key is `i.here` (lowercase), not `i.Here`. The PAPI and Dedot backends already handle this correctly — this was a PJS-only bug.

- **Fix summary:** Changed `i.Here` to `i.here === null` on line 700, consistent with the existing correct usage on line 444 (which already used lowercase). Updated the test fixtures to use lowercase `here` matching real PJS `toJSON()` behavior, and corrected the expected `destParaId` from `undefined` to `0`.

---

## ✅ Contributor Eligibility

- [x] My GitHub account's commit history is at least 6 months old.
- [x] I have set an on-chain identity on the Polkadot People Chain for my AssetHub address and requested a registrar judgement of at least `Reasonable` (verifiable via [Subscan](https://people-polkadot.subscan.io/)).
- [x] Every commit in this PR is signed (`-S`) and includes `Signed-off-by` and `AI-Assisted-By: <tool name(s)>` (or `AI-Assisted-By: None`) trailers.
- [x] I disclose any AI tools used to help write this fix (or confirm none were used), and I understand and can explain/defend the resulting code myself.
- [x] This fix was authored and reviewed by a human — not opened autonomously by an AI agent.
- [x] I will respond to maintainer review questions with specific, on-topic answers within 5 days.
- [x] I understand 🔴 High complexity fixes may require a brief live chat/call walkthrough before payout.
- [x] This is the only bug bounty issue I currently have reserved.

---

## ✅ Checklist

- [x] My code follows the project's code style.
- [x] I have added tests that prove my fix is effective (if applicable).
- [x] I have updated the documentation where necessary.
- [x] I have verified the fix does not introduce new issues.

---

## 💸 Polkadot Asset Hub Address (for Reward)

Polkadot Asset Hub Address: `16epYi2zGsDq6fsrJeJsUiFGtYfFxdHre5yPpHXdVcu84AbP`

---

## 🧩 Additional Notes

This is a 1-line code fix plus test fixture updates. The lowercase `here` key matches what PJS actually produces in `toJSON()` output, and is consistent with the existing `getDryRunCall` method on line 444 which already used the correct casing.

@michaeldev5

### CODE REVIEW

**Review by michaeldev5 — CHANGES_REQUESTED** [review]
Fix is correct. But the unit tests should also be fixed. Please also update the getDryRunXcm test to use { here: null } and expect destParaId: 0. Also rebase the branch on the latest main.
.

**Review by michaeldev5 — CHANGES_REQUESTED** [review]
Do not create multiple commits for this simple fix. Please make it one commit

File: packages/sdk-pjs/src/PolkadotJsApi.test.ts [hunk 1]
```diff
@@ -2055,7 +2055,7 @@ describe('PolkadotJsApi', () => {
   describe('getDryRunXcm', () => {
     const originLocation: TLocation = {
       parents: 0,
-      interior: { Here: null }
+      interior: { here: null }
```
Comment (michaeldev5) [inline packages/sdk-pjs/src/PolkadotJsApi.test.ts]: Please keep this as { Here: null }. This is a typed TLocation input, not PJS toJSON() output.

**Review by michaeldev5 — CHANGES_REQUESTED** [review]

**Review by michaeldev5 — APPROVED** [review]

**Review by dudo50 — APPROVED** [review]

**Comment by dudo50** [comment]
Thanks for your contribution @yaronski we have sent you the reward (Adjusted based on complexity of the PR):

https://assethub-polkadot.subscan.io/extrinsic/0x9947e25eea7a427c9bf36ee9876c059e1287476ad3b877d720a7902fdb423c4e

With kind regards,
Team ParaSpell✨
